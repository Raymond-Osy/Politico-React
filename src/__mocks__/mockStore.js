export const mockStoreData = {
  auth: {
    signupLoading: false,
    signupSuccess: false,
    singupFailure: false,
    signupError: {},
    loginLoading: false,
    loginSuccess: false,
    loginFailure: false,
    loginError: {}
  },
  parties: {
    partiesRequest: false,
    partiesSuccess: false,
    partiesFailure: false,
    partiesError: {},
    parties: []
  },
  toastr: {}
};
