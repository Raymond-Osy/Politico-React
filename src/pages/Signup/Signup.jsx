import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import ReduxToastr from 'react-redux-toastr';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Formik } from 'formik';
import * as Yup from 'yup';
import { signup } from '../../state/auth/action';
import './Signup.scss';
import TopNav from '../../components/Navigations/TopNav/TopNav';

export const Signup = props => {
  const formData = {
    firstname: '',
    lastname: '',
    othername: '',
    email: '',
    password: '',
    confirmPassword: '',
    phoneNumber: '09099999999',
    passportUrl: 'https://picsum.photos/200/300'
  };
  useEffect(() => {
    if (props.auth.signupSuccess === true) {
      props.history.push('/userProfile');
    }
  });
  return (
    <div className="custom-container signup-page">
      <ReduxToastr
        timeOut={5000}
        newestOnTop={false}
        preventDuplicates
        position="top-right"
        transitionIn="fadeIn"
        transitionOut="fadeOut"
        progressBar
        closeOnToastrClick/>
      <TopNav />
        <section className="container">
            <div className="row">
                <div className="main-info col-100 pt-0 bottom-space-15">
                    <h1><span>Signup</span></h1>
                </div>
            </div>
        </section>
        <footer className="footer-container-form">
          <Formik
            initialValues = {formData}
            onSubmit={values => props.signup(values)}
                validationSchema={Yup.object().shape({
                  firstname: Yup.string()
                    .required(),
                  lastname: Yup.string()
                    .required(),
                  othername: Yup.string()
                    .required(),
                  email: Yup.string()
                    .email()
                    .required('Required'),
                  password: Yup.string()
                    .required('No password provided.')
                    .min(5, 'Password is too short - should be 5 characters minimum.')
                    .matches(/(?=.*[0-9])/, 'Password must contain a number.'),
                  confirmPassword: Yup.string()
                    .required()
                    // eslint-disable-next-line func-names
                    .test('passwords-match', 'Passwords must match', function (value) {
                      return this.parent.password === value;
                    }),
                })}
            >
                  {data => {
                    const {
                      values, touched, errors, isSubmitting,
                      handleBlur, handleSubmit, handleChange
                    } = data;
                    return (
                        <form className="container" onSubmit={handleSubmit}>
                          <div className="form-container">
                              <label><b>First Name</b></label>
                              <input
                                type="text"
                                placeholder="Enter First Name"
                                name="firstname"
                                onChange={handleChange}
                                value={values.firstname}
                                onBlur={handleBlur}
                                className={errors.firstname && touched.firstname && 'error'}
                                required
                              />
                              {errors.firstname && touched.firstname && (
                                <div className="input-feedback">{errors.firstname}</div>
                              )}

                              <label><b>Last Name</b></label>
                              <input
                                type="text"
                                placeholder="Enter Last Name"
                                name="lastname"
                                onChange={handleChange}
                                value={values.lastname}
                                onBlur={handleBlur}
                                className={errors.lastname && touched.lastname && 'error'}
                                required
                              />
                                {errors.lastname && touched.lastname && (
                                <div className="input-feedback">{errors.lastname}</div>
                                )}

                              <label><b>Other Name</b></label>
                              <input
                                type="text"
                                placeholder="Enter Other Name"
                                name="othername"
                                onChange={handleChange}
                                value={values.thername}
                                onBlur={handleBlur}
                                className={errors.othername && touched.othername && 'error'}
                                required
                              />
                              {errors.othername && touched.othername && (
                                <div className="input-feedback">{errors.othername}</div>
                              )}

                              <label><b>Email Address</b></label>
                              <input
                                type="email"
                                placeholder="Enter Email Address"
                                name="email"
                                onChange={handleChange}
                                value={values.email}
                                onBlur={handleBlur}
                                className={errors.email && touched.email && 'error'}
                                required
                              />
                              {errors.email && touched.email && (
                                <div className="input-feedback">{errors.email}</div>
                              )}

                              <label><b>Password</b></label>
                              <input
                                type="password"
                                placeholder="Enter Password"
                                name="password"
                                onChange={handleChange}
                                value={values.password}
                                onBlur={handleBlur}
                                className={errors.password && touched.password && 'error'}
                                required
                              />
                              {errors.password && touched.password && (
                                <div className="input-feedback">{errors.password}</div>
                              )}

                              <label><b>Confirm Password</b></label>
                              <input
                                type="password"
                                placeholder="Confirm Password"
                                name="confirmPassword"
                                onChange={handleChange}
                                value={values.confirmPassword}
                                onBlur={handleBlur}
                                className={errors.confirmPassword && touched.confirmPassword && 'error'}
                                required
                                />
                                {errors.confirmPassword && touched.confirmPassword && (
                                  <div className="input-feedback">{errors.confirmPassword}</div>
                                )}

                              <button className="btn-form" type="submit" disabled={isSubmitting}>Signup</button>
                              <p className="text-center">Already have an account? login <Link to={'/login'} style={{ color: '#2F7A17', textDecoration: 'none' }}>here</Link></p>
                          </div>
                      </form>
                    );
                  }
                  }
            </Formik>
            <div className="footer-container-2">
              <p>Copyright &copy; 2019, Raymond Akalonu.</p>
            </div>
        </footer>
    </div>
  );
};

Signup.propTypes = {
  singup: PropTypes.func,
  auth: PropTypes.object
};

const mapStateToProps = state => ({
  auth: state.auth
});

export const ConnectedSignup = connect(mapStateToProps, { signup },)(Signup);
