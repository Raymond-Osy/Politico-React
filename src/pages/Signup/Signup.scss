@import "../Login/Login.scss";
.signup-page {
  .error-message {
      background-color: rgb(243, 205, 217);
      color:#530606;
      text-align:center;
      padding:10px 20px;
  }
  .success-message {
    background-color: rgb(205, 243, 221);
    color:#065339;
    text-align:center;
    padding:10px 20px;
  }
}
