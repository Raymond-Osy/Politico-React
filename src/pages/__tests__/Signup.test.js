import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import reduxStore from 'redux-mock-store';
import { cleanup, render } from 'react-testing-library';
import { mockStoreData } from '../../__mocks__/mockStore';
import { Signup } from '../Signup/Signup';

const mockStore = reduxStore();
const store = mockStore(mockStoreData);

const props = {
  auth: {
    signupError: {},
  }
};

afterEach(cleanup);

test('Should render the <signup /> component', () => {
  const SignupComponent = render(
    <Provider store={store}>
        <BrowserRouter>
            <Signup {...props}/>
        </BrowserRouter>
    </Provider>
  );
  expect(SignupComponent).toBeTruthy();
  expect(SignupComponent).toMatchSnapshot();
});

test('Renders <Signup /> form', () => {
  const { getByPlaceholderText } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Signup {...props} />
      </BrowserRouter>)
    </Provider>
  );
  expect(getByPlaceholderText('Enter First Name')).toBeTruthy();
  expect(getByPlaceholderText('Enter Last Name')).toBeTruthy();
  expect(getByPlaceholderText('Enter Other Name')).toBeTruthy();
  expect(getByPlaceholderText('Enter Email Address')).toBeTruthy();
  expect(getByPlaceholderText('Enter Password')).toBeTruthy();
  expect(getByPlaceholderText('Confirm Password')).toBeTruthy();
});
