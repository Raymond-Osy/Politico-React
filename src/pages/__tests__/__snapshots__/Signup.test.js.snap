// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the <signup /> component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="custom-container signup-page"
      >
        <nav
          class="row"
        >
          <div
            class="logo-div"
          >
            <a
              href="/"
            >
              <h1>
                Politico
              </h1>
            </a>
          </div>
          <div
            class="signup-signin-div"
          >
            <h4>
              <a
                href="/signup"
                style="text-decoration: none;"
              >
                Sign Up
              </a>
              <span
                class="m-side"
              >
                |
              </span>
              <a
                href="/login"
                style="text-decoration: none;"
              >
                Login
              </a>
            </h4>
          </div>
        </nav>
        <span
          aria-live="assertive"
          class="redux-toastr"
        >
          <span>
            <div
              class="top-left"
            />
            <div
              class="top-right"
            />
            <div
              class="top-center"
            />
            <div
              class="bottom-left"
            />
            <div
              class="bottom-right"
            />
            <div
              class="bottom-center"
            />
          </span>
        </span>
        <section
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="main-info col-100 pt-0 bottom-space-15"
            >
              <h1>
                <span>
                  Signup
                </span>
              </h1>
            </div>
          </div>
        </section>
        <footer
          class="footer-container-form"
        >
          <form
            class="container"
          >
            <div
              class="form-container"
            >
              <label>
                <b>
                  First Name
                </b>
              </label>
              <input
                name="firstname"
                placeholder="Enter First Name"
                required=""
                type="text"
                value=""
              />
              <label>
                <b>
                  Last Name
                </b>
              </label>
              <input
                name="lastname"
                placeholder="Enter Last Name"
                required=""
                type="text"
                value=""
              />
              <label>
                <b>
                  Other Name
                </b>
              </label>
              <input
                name="othername"
                placeholder="Enter Other Name"
                required=""
                type="text"
                value=""
              />
              <label>
                <b>
                  Email Address
                </b>
              </label>
              <input
                name="email"
                placeholder="Enter Email Address"
                required=""
                type="email"
                value=""
              />
              <label>
                <b>
                  Password
                </b>
              </label>
              <input
                name="password"
                placeholder="Enter Password"
                required=""
                type="password"
                value=""
              />
              <label>
                <b>
                  Confirm Password
                </b>
              </label>
              <input
                name="confirmPassword"
                placeholder="Confirm Password"
                required=""
                type="password"
                value=""
              />
              <button
                class="btn-form"
                type="submit"
              >
                Signup
              </button>
              <p
                class="text-center"
              >
                Already have an account? login 
                <a
                  href="/login"
                  style="color: rgb(47, 122, 23); text-decoration: none;"
                >
                  here
                </a>
              </p>
            </div>
          </form>
          <div
            class="footer-container-2"
          >
            <p>
              Copyright © 2019, Raymond Akalonu.
            </p>
          </div>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="custom-container signup-page"
    >
      <nav
        class="row"
      >
        <div
          class="logo-div"
        >
          <a
            href="/"
          >
            <h1>
              Politico
            </h1>
          </a>
        </div>
        <div
          class="signup-signin-div"
        >
          <h4>
            <a
              href="/signup"
              style="text-decoration: none;"
            >
              Sign Up
            </a>
            <span
              class="m-side"
            >
              |
            </span>
            <a
              href="/login"
              style="text-decoration: none;"
            >
              Login
            </a>
          </h4>
        </div>
      </nav>
      <span
        aria-live="assertive"
        class="redux-toastr"
      >
        <span>
          <div
            class="top-left"
          />
          <div
            class="top-right"
          />
          <div
            class="top-center"
          />
          <div
            class="bottom-left"
          />
          <div
            class="bottom-right"
          />
          <div
            class="bottom-center"
          />
        </span>
      </span>
      <section
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="main-info col-100 pt-0 bottom-space-15"
          >
            <h1>
              <span>
                Signup
              </span>
            </h1>
          </div>
        </div>
      </section>
      <footer
        class="footer-container-form"
      >
        <form
          class="container"
        >
          <div
            class="form-container"
          >
            <label>
              <b>
                First Name
              </b>
            </label>
            <input
              name="firstname"
              placeholder="Enter First Name"
              required=""
              type="text"
              value=""
            />
            <label>
              <b>
                Last Name
              </b>
            </label>
            <input
              name="lastname"
              placeholder="Enter Last Name"
              required=""
              type="text"
              value=""
            />
            <label>
              <b>
                Other Name
              </b>
            </label>
            <input
              name="othername"
              placeholder="Enter Other Name"
              required=""
              type="text"
              value=""
            />
            <label>
              <b>
                Email Address
              </b>
            </label>
            <input
              name="email"
              placeholder="Enter Email Address"
              required=""
              type="email"
              value=""
            />
            <label>
              <b>
                Password
              </b>
            </label>
            <input
              name="password"
              placeholder="Enter Password"
              required=""
              type="password"
              value=""
            />
            <label>
              <b>
                Confirm Password
              </b>
            </label>
            <input
              name="confirmPassword"
              placeholder="Confirm Password"
              required=""
              type="password"
              value=""
            />
            <button
              class="btn-form"
              type="submit"
            >
              Signup
            </button>
            <p
              class="text-center"
            >
              Already have an account? login 
              <a
                href="/login"
                style="color: rgb(47, 122, 23); text-decoration: none;"
              >
                here
              </a>
            </p>
          </div>
        </form>
        <div
          class="footer-container-2"
        >
          <p>
            Copyright © 2019, Raymond Akalonu.
          </p>
        </div>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
